x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/index.tsx

```diff
@@ -399,6 +402,21 @@ export function Detail() {
         pkgkey,
       });
 
+      const navigateOptions: InstallPkgRouteOptions =
+        onboardingAppId && onboardingLink
+          ? [
+              defaultNavigateOptions[0],
+              {
+                ...defaultNavigateOptions[1],
+                state: {
+                  ...(defaultNavigateOptions[1]?.state ?? {}),
+                  onCancelNavigateTo: [onboardingAppId, { path: onboardingLink }],
+                  onCancelUrl: onboardingLink,
+                },
+              },
+            ]
+          : defaultNavigateOptions;
+
       services.application.navigateToApp(...navigateOptions);
```

Comment:
This change is to allow customised redirect on cancel:



https://github.com/user-attachments/assets/88a197a1-1334-416b-b802-809008746f26